dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/GeneralWorkflowIT.java

```diff
@@ -70,6 +70,36 @@ public void resetDBBetweenTests() throws Exception {
         CommonTestUtilities.cleanStatePrivate2(SUPPORT, false);
     }
 
+    @Test
+    public void refreshAll() {
+        // get userid
+        final long userid = testingPostgres.runSelectStatement(String.format("SELECT id FROM user_profile WHERE username='%s';", USER_2_USERNAME), long.class);
+
+        // Make sure there are no entries associated with this user
+        testingPostgres.runDeleteStatement(String.format("DELETE FROM user_entry ue WHERE ue.userid = %d", userid));
+
+        // refresh all by organizations (We are testing the CLI implementation of this refresh functionality, we do the refresh here to give a baseline)
+        refreshByOrganizationReplacement(USER_2_USERNAME);
+
+        // initial count of workflows associated with this user using the standard refresh
+        final long val1 = testingPostgres.runSelectStatement(String.format("SELECT COUNT(*) FROM user_entry WHERE userid = %d;", userid), long.class);
+
+        // Delete all user_entry rows associated with repos this user is added to
+        final int deleteResult = testingPostgres.runDeleteStatement(String.format("DELETE FROM user_entry ue WHERE ue.userid = %d", userid));
+        assertEquals("After deletion, there should be 0 rows remaining associated with this user", deleteResult, val1);
+
+        // ensure resulting count after delete is less than the original count, otherwise we may not know if the final test is truly successful
+        final long val2 = testingPostgres.runSelectStatement(String.format("SELECT COUNT(*) FROM user_entry WHERE userid = %d;", userid), long.class);
+        assertTrue(String.format("There should be fewer entries post-deletion (pre-deletion, the user must be associated with > 0 workflows) pre: %d, post: %d", val1, val2), val1 > val2);
+
+        // run CLI refresh command to refresh all workflows
+        Client.main(new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "refresh"});
+
+        // final count of workflows associated with this user
+        final long val3 = testingPostgres.runSelectStatement(String.format("SELECT COUNT(*) FROM user_entry WHERE userid = %d;", userid), long.class);
```

Comment:
Because you control the data, and know exactly what you should get, you should assert the expected values. It sort of goes against the grain of writing generic code, but you should do it this way in tests.

For example, if you expect there to be 3 workflows after this (don't know if that's the case, just making up a number), then the end-result should be you are testing on line 100 that 3 workflows are expected.

Also, may not be applicable depending on the changes you make based on my above comment, but I feel it's better to have a well named variable rather than `val3`. You can even do away with comments in many cases.

```
final long userWorkflowCount = ...
```

Then when I look at that variable's usage on line 100, I can tell what it is, rather than having go back up a line and see what `val3` is. It's a small thing, especially in this context, where the variable is on two consecutive lines, but in general, it's easier to read code without having to back and forth.


+1 on making all your variables final.

Reply:
Updated the test to now directly checks for the expected 40 linked workflows to the test user. Variable names have been re-written to be more self-documenting.